src/Bicep.LangServer.IntegrationTests/Completions/SnippetTemplates/res-app-gateway/main.combined.bicep

```diff
@@ -0,0 +1,87 @@
+resource applicationGateway 'Microsoft.Network/applicationGateways@2020-11-01' = {
+  name: 'name'
+  location: resourceGroup().location
+  properties: {
+    sku: {
+      name: 'Standard_Small'
+      tier: 'Standard'
+      capacity: 2
+    }
+    gatewayIPConfigurations: [
+      {
+        name: 'name'
+        properties: {
+          subnet: {
+            id: resourceId('Microsoft.Network/virtualNetworks/subnets', 'virtualNetwork', 'Subnet-1')

```

Comment:
Should we encourage symbolic references to other resources rather than using the `resourceId()` function in snippets?

Reply:
@anthony-c-martin, I don't think we have templates checked in for most of the 'resourceProvider/type' listed in resourceId(..).

I had an offline discussion with @MarcusFelling  regarding this. I am going to update it to be a tab stop instead. Do you have any concerns?

Reply:
@bhsubra - maybe this could be a topic in the next DSL discussion? I'd definitely like to make sure we're encouraging best practices with the snippets, as they're going to be an entry-point to the language and serve as a learning opportunity for many new users.

Reply:
We're not disagreeing with that. I suggested we remove the use of resourceId function in all snippets and instead have a tab stop/placeholder where symbolic name can be used. We can't reference another resource's symbolic link in a single resource snippet.

Reply:
Updated to use placholders here:
https://github.com/Azure/bicep/commit/ea170a6cefb1cfd227d5ffb7f8242f63aa441032